src/main/java/poChat/bot/Parser.java

```diff
@@ -0,0 +1,167 @@
+                if (taskDescription.isEmpty()) {
+                    throw new TaskDescriptionEmptyException();
+                }
+                addToListAndReply(taskDescription);
+            } catch (StringIndexOutOfBoundsException e) {
+                throw new TaskDescriptionEmptyException();
+            }
+        } else if (textInput.startsWith("deadline")) {
+            try {
+                int byIndex = textInput.indexOf("/by");
+
+                String taskDescription = textInput.substring(9, byIndex - 1);
+                String deadline = textInput.substring(byIndex + 4);
+
+                if (taskDescription.isEmpty()) {
+                    throw new TaskDescriptionEmptyException();
+                }
+
+                addToListAndReply(taskDescription, deadline);
+            } catch (StringIndexOutOfBoundsException e) {
+                throw new TaskDescriptionEmptyException();
+            }
+        } else if (textInput.startsWith("event")) {
+            try {
+                int fromIndex = textInput.indexOf("/from");
+                int toIndex = textInput.indexOf("/to");
+
+                String taskDescription = textInput.substring(6, fromIndex - 1);
+                String startDate = textInput.substring(fromIndex + 6, toIndex - 1);
+                String endDate = textInput.substring(toIndex + 4);
+
+                if (taskDescription.isEmpty()) {
+                    throw new TaskDescriptionEmptyException();
+                }
+
+                addToListAndReply(taskDescription, startDate, endDate);
+            } catch (StringIndexOutOfBoundsException e) {
+                throw new TaskDescriptionEmptyException();
+            }
+        } else if (textInput.startsWith("find")) {
+            String keyword = textInput.substring(5);
+            this.taskList.printMatchingTasks(keyword);
+        } else {
+            replyToInvalidInput();
+        }
+
+        return isChatOver;
+    }
+
+    /**
+     * Prints the goodbye message to the user
+     */
+    private void sayGoodbye() {
+        String goodbyeMessage = "Bye. Hope to see you again soon!";
+        System.out.println(goodbyeMessage);
+    }
+
+    private void addTaskToList(Task task) {
+        taskList.add(task);
+        System.out.println("Got it. I've added this task:\n" + task + "\nNow you have "
+                + this.getNumTasks() + " pochat.tasks in the list.");
+    }
+
+    private void addToListAndReply(String textInput) {
+        Task task = new ToDo(textInput);
+        this.addTaskToList(task);
+    }
+
+    private void addToListAndReply(String textInput, String deadline) {
+        Task task = new Deadline(textInput, deadline);
+        this.addTaskToList(task);
+    }
+
+    private void addToListAndReply(String textInput, String startDate, String endDate) {
+        Task task = new Event(textInput, startDate, endDate);
+        this.addTaskToList(task);
+    }
+
+    private void replyWithListOfTextsEntered() {
```

Comment:
Maybe you can consider shortening this to replyWithListOfTexts or replyWithList? So that it's slightly shorter